extensions/blocks/contact-form/edit.js

```diff
@@ -239,78 +237,124 @@ class JetpackContactForm extends Component {
+											onKeyDown={ openOnArrowDown }
+											icon={ <Icon icon="edit" /> }
+										/>
+									);
+								} }
+								renderContent={ () => this.renderFormSettings() }
+							/>
+						</ToolbarGroup>
+					</BlockControls>
+				) }
+
 				<InspectorControls>
-					<PanelBody title={ __( 'Email Feedback Settings', 'jetpack' ) }>
-						{ this.renderToAndSubjectFields() }
-					</PanelBody>
-					<PanelBody title={ __( 'Confirmation Message', 'jetpack' ) }>
-						{ this.renderConfirmationMessageFields() }
+					<PanelBody title={ __( 'Form Settings', 'jetpack' ) }>
+						{ this.renderFormSettings() }
 					</PanelBody>
 				</InspectorControls>
+
 				<div className={ formClassnames }>
-					{ ! hasFormSettingsSet && (
-						<Placeholder
-							label={ __( 'Form', 'jetpack' ) }
-							icon={ renderMaterialIcon(
-								<Path d="M13 7.5h5v2h-5zm0 7h5v2h-5zM19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM11 6H6v5h5V6zm-1 4H7V7h3v3zm1 3H6v5h5v-5zm-1 4H7v-3h3v3z" />
-							) }
-							instructions={ __(
-								'You’ll receive an email notification each time someone fills out the form. Where should it go, and what should the subject line be?',
-								'jetpack'
-							) }
-						>
-							<form onSubmit={ this.onFormSettingsSet }>
-								{ this.renderToAndSubjectFields() }
-								<div class="components-placeholder__instructions">
-									{ __(
-										'(If you leave these blank, notifications will go to the author with the post or page title as the subject line.)',
-										'jetpack'
-									) }
-								</div>
-								<div className="jetpack-contact-form__create">
-									<Button isLarge isSecondary type="submit" disabled={ this.hasEmailError() }>
-										{ __( 'Add form', 'jetpack' ) }
-									</Button>
-								</div>
-							</form>
-						</Placeholder>
-					) }
-					{ hasFormSettingsSet && (
-						<InnerBlocks
-							allowedBlocks={ ALLOWED_BLOCKS }
-							templateLock={ false }
-							template={ [
-								[
-									'jetpack/field-name',
-									{
-										required: true,
-									},
-								],
-								[
-									'jetpack/field-email',
-									{
-										required: true,
-									},
-								],
-								[ 'jetpack/field-url', {} ],
-								[ 'jetpack/field-textarea', {} ],
-							] }
-						/>
-					) }
-					{ hasFormSettingsSet && <SubmitButton { ...this.props } /> }
+					<InnerBlocks allowedBlocks={ ALLOWED_BLOCKS } templateInsertUpdatesSelection={ false } />
 				</div>
-			</Fragment>
+			</>
 		);
 	}
 }
 
-export default compose( [ withInstanceId ] )( JetpackContactForm );
+export default compose( [
+	withSelect( ( select, props ) => {
+		const { getBlockType, getBlockVariations, getDefaultBlockVariation } = select( 'core/blocks' );
+		const { getBlocks } = select( 'core/block-editor' );
+		const { getSite } = select( 'core' );
+		const { replaceInnerBlocks, selectBlock } = useDispatch( 'core/block-editor' );
```

Comment:
Never seen `use` hooks inside a `with` HOC. 🤔 
I'm tempted to suggest using the `useSelect` hook as well, and keeping `useDispatch` separate, but I'm probably overthinking it.

Reply:
Switched this to use `withDispatch` instead. Good catch!